Assets/_BForBoss/_Weapons/Scripts/Bullets/BulletSpawner.cs

```diff
@@ -56,8 +58,16 @@ private void SetupPools()
             }
         }
 
+        private void Awake()
+        {
+            _wallHitVFXSpawner = GetComponent<WallHitVFXSpawner>();
+        }
+
         private void OnRelease(BulletBehaviour bullet)
         {
+            var wallHit = _wallHitVFXSpawner.SpawnWallHitVFX();
+            wallHit.transform.position = bullet.transform.position;
+            wallHit.Spawn();
```

Comment:
I'm a bit confused by this, doesn't this mean the wallHit is getting spawned twice?

Reply:
I was under the impression that the `SpawnWAllHitVFX` get it, then the `Spawn` starts the VFX. 
Yeah this is a bit confusing, i'll change it to be more intuitive.

Reply:
I get that part I think, but isn't `Spawn` itself getting called twice, since its still in `WeaponBehavior.OnBulletHitWall`?

Reply:
`WeaponBehavior.OnBulletHitWall` i'll double check this. Sinec it's different logic between raycast vs projectile

Reply:
@unoctium1 Okay I checked, and it doesn't get ccalled twice. 
I was going to refactor, but that would be too much